appserver/payara-appserver-modules/security-oauth2/src/main/java/fish/payara/security/oauth2/OAuth2AuthenticationMechanism.java

```diff
@@ -108,44 +114,77 @@ public OAuth2AuthenticationMechanism() {
 
     /**
      * Creates an OAuth2AuthenticationMechanism that has been defined using an annotation
-     * @param definition 
+     *
+     * @param definition
      */
     public OAuth2AuthenticationMechanism(OAuth2AuthenticationDefinition definition) {
         setDefinition(definition);
 
     }
 
     /**
-     * Sets the properties of the OAuth2AuthenticationMechanism as defined in an
-     * {@link OAuth2AuthenticationDefinition} annotation.
+     * Sets the properties of the OAuth2AuthenticationMechanism as defined in an {@link OAuth2AuthenticationDefinition} annotation.
+     *
      * @param definition
-     * @return 
+     * @return
      */
-    public OAuth2AuthenticationMechanism setDefinition(OAuth2AuthenticationDefinition definition) {;
+    public OAuth2AuthenticationMechanism setDefinition(OAuth2AuthenticationDefinition definition) {
```

Comment:
I think you could reduce the complexity of this method with a utility method that does the check of which config to use if present. Especially as they are all Strings.
something like 
authConfig = getConfig(definition.authEndpoint())